compiler/rustc_mir_build/src/thir/cx/expr.rs

```diff
@@ -902,8 +902,17 @@ impl<'tcx> Cx<'tcx> {
                     let alloc_id = self.tcx.create_static_alloc(id);
                     ExprKind::StaticRef { alloc_id, ty, def_id: id }
                 };
-                ExprKind::Deref {
-                    arg: self.thir.exprs.push(Expr { ty, temp_lifetime, span: expr.span, kind }),
+                if self.is_custom_mir {
+                    kind
+                } else {
+                    ExprKind::Deref {
+                        arg: self.thir.exprs.push(Expr {
+                            ty,
+                            temp_lifetime,
+                            span: expr.span,
+                            kind,
+                        }),
+                    }
                 }
```

Comment:
I don't like this too much, can we revert it once `&Static(Name)` is a thing?

Reply:
Reverted the changes to this part of the code and pushed an additional commit that has the alternative to this. Should be much better, not sure why I didn't do this in the first place

Reply:
yea, I like this.